relayer/src/metrics/para.rs

```diff
@@ -0,0 +1,225 @@
+        self
+    }
+
+    pub async fn build(self) -> AnyResult<MetricsCollector> {
+        let client = self
+            .client
+            .expect("client is not specified. It's developer mistake");
+        let network_id = self
+            .network_id
+            .expect("client is not specified. It's developer mistake");
+
+        Ok(MetricsCollector { client, network_id })
+    }
+}
+
+pub struct MetricsCollector {
+    client: SubUnsignedClient<ParachainConfig>,
+    network_id: GenericNetworkId,
+}
+
+impl MetricsCollector {
+    pub async fn update_metric<S, F>(&self, address: &S, block_hash: H256, update_fn: F)
+    where
+        S: StorageAddress<IsFetchable = Yes>,
+        F: Fn(<S::Target as DecodeWithMetadata>::Target),
+    {
+        match self.client.storage_fetch(address, block_hash).await {
+            Ok(Some(value)) => {
+                update_fn(value);
+            }
+            Err(err) => {
+                log::warn!("Failed to get value from storage: {err}");
+            }
+            Ok(None) => {}
+        }
+    }
+
+    pub async fn run(self) -> AnyResult<()> {
+        let mut interval = tokio::time::interval(ParachainConfig::average_block_time());
+        loop {
+            interval.tick().await;
+            let Ok(block_hash) = self.client.block_hash(()).await else {
+                continue;
+            };
+
+            let address = parachain_runtime::storage().beefy_mmr().beefy_authorities();
+            self.update_metric(&address, block_hash, |vset| {
+                metrics::absolute_counter!(PARA_BEEFY_CURRENT_ID, vset.id);
+                metrics::absolute_counter!(PARA_BEEFY_CURRENT_LEN, vset.len as u64);
+            })
+            .await;
+
+            let address = parachain_runtime::storage()
+                .beefy_mmr()
+                .beefy_next_authorities();
+
+            self.update_metric(&address, block_hash, |vset| {
+                metrics::absolute_counter!(PARA_BEEFY_NEXT_ID, vset.id);
+                metrics::absolute_counter!(PARA_BEEFY_NEXT_LEN, vset.len as u64);
+            })
+            .await;
+
+            match self.network_id {
+                GenericNetworkId::EVM(_) => {
+                    unimplemented!("EVM bridge is not supported in parachain");
+                }
+                GenericNetworkId::Sub(network_id) => {
+                    let labels = &[("network_id", format!("{:?}", network_id))];
+
+                    let address = parachain_runtime::storage()
+                        .beefy_light_client()
+                        .current_validator_set(network_id);
+                    self.update_metric(&address, block_hash, |vset| {
+                        metrics::absolute_counter!(
+                            PARA_BEEFY_LIGHT_CLIENT_CURRENT_ID,
+                            vset.id,
+                            labels
+                        );
+                        metrics::absolute_counter!(
```

Comment:
Counter is a monotonic value, but as I understand, the next length might be less than current. Consider `gauge` type for the metric.